cylc/flow/scheduler.py

```diff
@@ -307,6 +306,11 @@ def __init__(self, reg: str, options: Values) -> None:
             pub_d=os.path.join(self.workflow_run_dir, 'log')
         )
         self.is_restart = Path(self.workflow_db_mgr.pri_path).is_file()
+
+        # In the case of a restart get UUID from DB:
+        if not self.is_restart:
+            self.uuid_str = str(uuid4())
```

Comment:
Possibly we will just have to load the UUID from the DB here if it's a restart

Reply:
That's what I did originally, and Oliver Objected to.

Reply:
> That's what I did originally, and Oliver Objected to.

Oliver objected to it being done two ways in two places with potential for mixups.

> But still need to carefully check the implications of moving self.load_workflow_params_and_tmpl_vars()

Yes, very careful!